src/desktop_app_info.rs

```diff
@@ -36,3 +55,177 @@ impl DesktopAppInfo {
         }
     }
 }
+
+#[cfg(any(feature = "v2_58", feature = "dox"))]
+fn launch_uris_as_manager_with_fds_inner<P: IsA<AppLaunchContext>, O: IsA<DesktopAppInfo>>(
+    this: &O,
+    uris: &[&str],
+    launch_context: Option<&P>,
+    spawn_flags: glib::SpawnFlags,
+    user_setup: Option<Box_<dyn FnOnce() + 'static>>,
+    pid_callback: Option<&mut dyn (FnMut(&DesktopAppInfo, glib::Pid))>,
+    stdin_fd: FD,
+    stdout_fd: FD,
+    stderr_fd: FD,
```

Comment:
This should become a generic function with `FD: IntoRawFd`, or be marked as `unsafe`

Reply:
Why `IntoRawFd` and not `AsRawFd`?

Reply:
You give ownership of the fds to the function or not?

Reply:
I wouldn't, that's why I pcked `As*` instead of `Into*`.

Reply:
It doesn't matter what you would do :P What does the C function do?

Reply:
It appears that the FDs need to be closed outside of the function: https://github.com/GNOME/gnome-shell/blob/master/src/shell-app.c#L1322

Therefore, we don't give ownership.

Reply:
Then you're right :)